titiler/api/endpoints/mosaic.py

```diff
@@ -0,0 +1,245 @@
+"""API for MosaicJSON Dataset."""
+import asyncio
+import os
+import random
+from functools import partial
+from io import BytesIO
+from typing import Callable, Sequence
+
+import mercantile
+import morecantile
+import numpy
+from cogeo_mosaic.backends import MosaicBackend
+from cogeo_mosaic.mosaic import MosaicJSON
+from cogeo_mosaic.utils import get_footprints
+from rasterio.transform import from_bounds
+from rio_tiler.io.cogeo import tile as cogeoTiler
+from rio_tiler.profiles import img_profiles
+from rio_tiler.utils import render
+
+from titiler.api.deps import CommonTileParams, TileMatrixSetNames
+from titiler.api.endpoints.cog import cog_info, tile_response_codes
+from titiler.api.utils import postprocess
+from titiler.models.metadata import cogBounds, cogInfo
+from titiler.models.mosaic import CreateMosaicJSON, UpdateMosaicJSON
+from titiler.ressources.common import drivers
+from titiler.ressources.enums import ImageMimeTypes, ImageType, PixelSelectionMethod
+from titiler.ressources.responses import ImgResponse
+
+from fastapi import APIRouter, Depends, Path, Query
+from fastapi.routing import APIRoute
+
+from starlette.concurrency import run_in_threadpool
+from starlette.exceptions import HTTPException
+from starlette.requests import Request
+from starlette.responses import Response
+
+
+def _chunks(my_list: Sequence, chuck_size: int):
+    """Yield successive n-sized chunks from l."""
+    for i in range(0, len(my_list), chuck_size):
+        yield my_list[i : i + chuck_size]
+
+
+class MosaicJSONRouter(APIRoute):
+    """Custom router to temporarily forbid usage of cogeo-mosaic STAC backend"""
+
+    def get_route_handler(self) -> Callable:
+        """Override base method (https://fastapi.tiangolo.com/advanced/custom-request-and-route/)"""
+        original_route_handler = super().get_route_handler()
+
+        async def forbid_stac_backend(request: Request) -> Response:
+            url = request.query_params.get("url")
+            if not url:
+                body = await request.json()
+                url = body.get("url")
+            if url.startswith("stac"):
+                # Raise as a validation error
+                raise HTTPException(
+                    status_code=422,
+                    detail="STAC is not currently supported for mosaicjson endpoints",
+                )
+            return await original_route_handler(request)
+
+        return forbid_stac_backend
+
+
+router = APIRouter()
+
+
+@router.post(
+    "", response_model=MosaicJSON, response_model_exclude_none=True,
+)
+def create_mosaicjson(body: CreateMosaicJSON):
+    """Create a MosaicJSON"""
+    mosaic = MosaicJSON.from_urls(
+        body.files,
+        minzoom=body.minzoom,
+        maxzoom=body.maxzoom,
+        max_threads=body.max_threads,
+    )
+    with MosaicBackend(body.url, mosaic_def=mosaic) as mosaic:
```

Comment:
I'm kinda hesitant with letting user write where they want. In an ideal world the mosaic id should be something like `username.layer` (like mapbox tileset https://github.com/mapbox/pxm-manifest-specification/blob/master/schemas/pxm-manifest-0.5.1.json#L22)

Reply:
by default the app should either write to efs or to s3 IMO

Reply:
Off the top of my head those are the only backends which support write.  Whats missing (and will be added later) is backends which don't support `write` (i.e. raise `NotImplementedError`) will raise some sort of exception.

I've just been using S3 backend while developing because it supports basically everything.

Reply:
Resolving in favor of https://github.com/developmentseed/titiler/issues/3